### DISCUSSION TITLE
How to get the properties of an object in HFSS 3D Layout?

### DISCUSSION THREAD

**Discussion by kruglov-dmitrii** [ansys/pyaedt#4560] [Q&A] [answered]
Hi!

How can I can read the properties of objects in 3DLayout? I would like to, for example, read the positions of all the vias in my design.

I can list all the vias in my design, or change their properties, but I cannot find anything like `p.get_properties` method.
```
test = Hfss3dLayout(designname='mydesign')
p = test.modeler
p.vias # -> lists all vias
p.change_property('via_591', property_name='HoleDiameter', property_value=0.15) # -> edit a selected via
```

How can I do something like this?
```
via_pos = p.get_properties('via_591', 'Location')
```

Thanks!

D.

**Comment by Samuelopez-ansys** [comment, marked as answer]
Hi @kruglovIFX ,

For Layout automation and manipulation, I encourage you to use PyEDB, because you do not need HFSS 3DLayout until the solver part and postprocessing.

Take a look:

https://edb.docs.pyansys.com/version/stable/index.html

For example, you can have a full control of the pad stack:

https://edb.docs.pyansys.com/version/stable/api/PadstackEdb.html

**Reply by Samuelopez-ansys** [reply]
To use EDB from an existing HFSS3DLayout, you just need to use the .aedb directory created when you create an HFSS 3DLayout design. 
PyEDB also allows other ECAD formats.
